pom.xml

```diff
@@ -277,20 +277,20 @@
         <plugin>
           <groupId>org.apache.maven.plugins</groupId>
           <artifactId>maven-javadoc-plugin</artifactId>
-          <version>2.10.4</version>
+          <version>3.0.1</version>
```

Comment:
this version has an issue with doing a multiproject build, that's why we are still using the old version

Reply:
Can you provide a way to reproduce this issue.

Reply:
IIRC a simple

mvn javadoc:aggregate

should already fail

Reply:
Yes that fails with:

~~~
ERROR] Failed to execute goal org.apache.maven.plugins:maven-dependency-plugin:3.0.2:unpack-dependencies (copy_demo_certs) on project hono-core: Artifact has not been packaged yet. When used on reactor artifact, unpack should be executed after packaging: see MDEP-98. -> [Help 1]
~~~

But I think this is a bug in the Hono build which simply didn't manifest until now in that plugin (but in the Eclipse IDE) and which can be prevented by:

~~~
mvn package javadoc:aggregate
~~~

 I think the proper way to fix this would be make of use of the [remote resources plugin](https://maven.apache.org/plugins/maven-remote-resources-plugin/).

So would this workaround above the fine for the moment? Or should I try to replace the "dependency" plugin with the "remove resources" plugin? As I am pretty sure the Maven Javadoc Plugin will not change their behavior.

Reply:
No, I think we can live that shortcoming. We are using the aggregate goal in our release builds but we always use it in conjunction with a another goal like *install*, so it shouldn't be a problem.